src/geotiffimage.js

```diff
@@ -257,8 +257,10 @@ class GeoTIFFImage {
       offset = this.fileDirectory.StripOffsets[index];
       byteCount = this.fileDirectory.StripByteCounts[index];
     }
-    const slice = await this.source.fetch(offset, byteCount);
-
+    const slice = await this.source.fetch(offset, byteCount, false, signal);
+    if (slice.length === 0) {
```

Comment:
Maybe use
```js
try {

} catch (AbortError) {

}
```
? Is checking `length` preferable?

Reply:
This I was no sure of what to do.  Since the exception below happens within the `set` method, the error doesn't propagate upwards.  So we can't catch the `AbortError` here - we could check `signal.aborted` though.